From d719945c19e030bbac75a068fb61ec4ec07a511d Mon Sep 17 00:00:00 2001
From: Jiang Lu <lu.jiang@windriver.com>
Date: Thu, 6 Jun 2019 14:45:41 +0800
Subject: [PATCH 5/8] usb:otg_fsm:fix a compile issue

Commit 9b1635e8523226bf679a08c94898fe59bc55d2ee(USB: check
usb_get_extra_descriptor for proper size) add a parameter
for usb_get_extra_descriptor(). Adjust parameters when invoking
usb_get_extra_descriptor() in otg_fsm driver accordingly.

Signed-off-by: Jiang Lu <lu.jiang@windriver.com>
---
 drivers/usb/common/usb-otg-fsm.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/common/usb-otg-fsm.c b/drivers/usb/common/usb-otg-fsm.c
index f0538f5b99fa..626f4ffbe562 100644
--- a/drivers/usb/common/usb-otg-fsm.c
+++ b/drivers/usb/common/usb-otg-fsm.c
@@ -158,7 +158,8 @@ static void otg_hnp_polling_work(struct work_struct *work)
 	if (fsm->tst_maint &&
 		(__usb_get_extra_descriptor(udev->rawdescriptors[0],
 		le16_to_cpu(udev->config[0].desc.wTotalLength),
-				USB_DT_OTG, (void **) &desc) == 0)) {
+				USB_DT_OTG, (void **) &desc,
+				sizeof(*desc)) == 0)) {
 		/* shorter bLength of OTG 1.3 or earlier */
 		if (desc->bLength < 5) {
 			fsm->a_bus_req = 0;
-- 
2.17.1

